app/2.0/docs/tools/build-for-production.md

```diff
@@ -0,0 +1,211 @@
+
+With HTTP/2 push, support for the PRPL pattern is possible:
+
+* Push critical resources for the initial route.
+* Render initial route.
+* Pre-cache remaining routes.
+* Lazy-load and create remaining routes on demand.
+
+This pattern requires an unbundled build (the default build type).
+
+Because not all browsers support HTTP/2 push, you will also need to create a bundled build:
+
+   ```bash
+   polymer build --bundle
+   ```
+   
+### Polyfills?
+
+**TODO: Elaborate on why/how this is part of the build process**
+
+* Use webcomponents-loader.js? 
+* Full polyfill? 
+* Add them differentially?
+
+## Building with the CLI
+
+Run `polymer help build` to see the command line options for 
+
+You can define your build options by editing the `builds` object in your project's `polymer.json` configuration file. For detailed information on `polymer.json`, see the [`polymer.json` specification](https://www.polymer-project.org/2.0/docs/tools/polymer-json).
+
+### Example 1: A single, compiled, bundled build
+
+This example gives a single build. ES6 is compiled to ES5; JavaScript, CSS and HTML are minified; and resources are bundled.
+
+`polymer.json` {.caption}
+
+```json
+...
+"builds": [{
+  "bundle": true,
+  "js": {"compile": true, "minify": true},
+  "css": {"minify": true},
+  "html": {"minify": true}
+}]
+...
+```
+
+To build this configuration:
+
+1. Edit your `polymer.json` file to include the build configuration above.
+2. Ensure that you have installed the latest version of the [Polymer CLI](https://www.polymer-project.org/2.0/docs/tools/polymer-cli).
+3. cd to your project's main folder, and type `polymer build`.
+
+Your build is output to the `build/default` folder.
+
+### Example 2: Multiple builds, both bundled and unbundled
+
+This example gives two builds - bundled and unbundled.
+
+`polymer.json` {.caption}
+
+```json
+...
+"builds": [
+  {
+    "name": "bundled",
+    "bundle": true,
+    "js": {"minify": true},
+    "css": {"minify": true},
+    "html": {"minify": true}
+  },
+  {
+    "name": "unbundled",
+    "js": {"minify": true},
+    "css": {"minify": true},
+    "html": {"minify": true}
+  }
+]
+...
+```  
+
+To build this configuration:
+
+1. Edit your `polymer.json` file to include the build configuration above.
+2. Ensure that you have installed the latest version of the [Polymer CLI](https://www.polymer-project.org/2.0/docs/tools/polymer-cli).
+3. cd to your project's main folder, and type `polymer build`.
+
+Your builds are output to two separate folders, corresponding to their names: `build/bundled` and `build/unbundled`. 
+
+## Building with `polymer-build` {#use-polymer-build}
+
+Consider using `polymer-build` instead of the CLI if you:
+
+* Want to customize your build(s) without using the Polymer CLI
+* Need to run your source code through custom optimizers/processors before, after, or during your build
+* Need to hook additional work into any part of the build process
```

Comment:
For now, I've used https://github.com/PolymerElements/generator-polymer-init-custom-build/blob/master/generators/app/gulpfile.js. If I can get a better example, will update.